### PR TITLE
fix(lessons): precheck GEPA duplicates

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary
- honor existing duplicate-check flags in  before running GEPA
- skip GEPA work when a similar lesson is found and duplicate skipping is enabled
- add regression tests for skip-vs-warn duplicate behavior

## Testing
- ..                                                                       [100%]
2 passed in 0.35s
- All checks passed!

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
Closing in favor of #936.

PR #936 (Codex, session 5dbf) does the same thing but better:
- Extracts shared helpers (`_prepare_existing_lesson_check`, `_check_existing_lesson_similarity`) 
- Refactors both `generate_lessons_with_evolution()` AND `generate_lessons_from_analysis()` to use them (DRY)
- Uses `click.echo()` consistently (the Greptile concern about silent fallback on `verbose=False` is resolved — the shared helper always outputs warnings)
- 100% patch coverage with 5 tests covering missing-dir and disabled-check edge cases

No point shipping the subset when the superset is already written and more thorough. Promoting #936 from draft next.

File: packages/gptme-lessons-extras/src/gptme_lessons_extras/generate.py [hunk 1]
```diff
@@ -135,6 +135,19 @@ def generate_lessons_with_evolution(
...
+        if existing_lessons_dir is None:
+            existing_lessons_dir = Path("lessons")
+
+        if not existing_lessons_dir.exists():
+            if verbose:
+                print(
+                    f"  Warning: Existing lessons directory not found: {existing_lessons_dir}"
+                )
+                print("  Skipping duplicate check.")
+            check_existing = False
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-lessons-extras/src/gptme_lessons_extras/generate.py:156]: Fixed in 609ddcb. `generate_lessons_with_evolution()` now surfaces the missing-`existing_lessons_dir` warning without `-v`, and I made the duplicate-precheck messages visible in the non-verbose path to match `generate_lessons_from_analysis()`. Added regression coverage for the missing-dir case and the warn-only duplicate flow.

File: packages/gptme-lessons-extras/src/gptme_lessons_extras/generate.py [hunk 2]
```diff
@@ -151,8 +164,35 @@ def generate_lessons_with_evolution(
...
+                top_match = similar_lessons[0]
+                similarity_pct = int(top_match["similarity"] * 100)
+                if verbose:
+                    print(
+                        f"  ⚠️  Similar lesson found ({similarity_pct}% match): {top_match['title']}"
+                    )
+                    print(
+                        "     Location: "
+                        f"{top_match['filepath'].relative_to(existing_lessons_dir)}"
+                    )
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-lessons-extras/src/gptme_lessons_extras/generate.py]: Fixed in 609ddcb. I added `_format_lesson_location()` so duplicate reporting falls back to the raw path when the match path is outside `existing_lessons_dir`, and reused it in both generator paths. Added a regression test that returns an out-of-tree match path and verifies we still report it cleanly.
